### PR TITLE
Bug/object reference leak

### PR DESCRIPTION
**Pull request by oliverjohnstone**

### CODE REVIEW

File: test/update.js [hunk 1]
```diff
@@ -123,6 +122,28 @@ module.exports = function(idProperty, getEngine) {
...
+            savedObject.newProperty = true
+            newObject.should.not.have.property('newProperty')
+            done()
+          })
+
+          delete newObject.newProperty
+
+          engine.update(newObject, false, function (error, savedObject) {
+            savedObject.newProperty = true
+            newObject.should.not.have.property('newProperty')
```
Comment (serby) [inline test/update.js]: I'm not sure this assert will be called?

File: test/update.js [hunk 2]
```diff
@@ -123,6 +122,28 @@ module.exports = function(idProperty, getEngine) {
       })
     })
 
+    it('should always return a new object', function (done) {
```
Comment (serby) [inline test/update.js]: I would like to see a test that describes problem that we've seen more clearly.

`should return a unreferenced overridden object when override is true`

**Comment by oliverjohnstone** [comment]
@serby good spot, thanks. I've made those changes.

**Comment by serby** [comment]
:+1: Looks nice. Thanks

**Pull request merged**